src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -513,6 +521,145 @@ export class WorkspacesUsecases {
     return teamsWithMemberCount;
   }
 
+  async getTeamMembers(teamId: WorkspaceTeam['id']) {
+    const members = await this.teamRepository.getTeamMembers(teamId);
+    return members;
+  }
+
+  async editTeamData(teamId: WorkspaceTeam['id'], editTeamDto: EditTeamDto) {
+    const [team, workspace] = await Promise.all([
+      this.teamRepository.getTeamById(teamId),
+      this.workspaceRepository.findOne({ defaultTeamId: teamId }),
+    ]);
+
+    if (!team || workspace?.isDefaultTeam(team)) {
+      throw new BadRequestException('Invalid team');
+    }
+
+    await this.teamRepository.updateById(teamId, editTeamDto);
+  }
+
+  async removeMemberFromTeam(
+    teamId: WorkspaceTeam['id'],
+    memberId: User['uuid'],
+  ) {
+    const [team, workspace] = await Promise.all([
+      this.teamRepository.getTeamById(teamId),
+      this.workspaceRepository.findOne({ defaultTeamId: teamId }),
+    ]);
+
+    if (!team || workspace?.isDefaultTeam(team)) {
+      throw new BadRequestException('Invalid team');
+    }
+
+    const teamUser = await this.teamRepository.getTeamUser(memberId, team.id);
+
+    if (!teamUser) {
+      throw new BadRequestException('User is not part of team!');
+    }
+
+    const isUserManagerOfTeam = team.managerId === memberId;
+
+    if (isUserManagerOfTeam) {
+      const teamWorkspace = await this.workspaceRepository.findOne({
+        id: team.workspaceId,
+      });
+
+      await this.teamRepository.updateById(team.id, {
+        managerId: teamWorkspace.ownerId,
+      });
+    }
+
+    await this.teamRepository.removeMemberFromTeam(teamId, memberId);
+  }
+
+  async addMemberToTeam(teamId: WorkspaceTeam['id'], memberId: User['uuid']) {
+    const [team, workspace] = await Promise.all([
+      this.teamRepository.getTeamById(teamId),
+      this.workspaceRepository.findOne({ defaultTeamId: teamId }),
+    ]);
+
+    if (!team || workspace?.isDefaultTeam(team)) {
```

Comment:
Same issue as with `editTeamData`. As this check seems to be repetitive, I suggest to create a function just for this